static/src/javascripts/projects/commercial/modules/dfp/render-advert.ts

```diff
@@ -237,16 +228,16 @@ export const renderAdvert = (
 					 * */
 					advert.hasPrebidSize = false;
 
+					const sizeCallback = sizeCallbacks[size];
 					return Promise.resolve(
-						sizeCallbacks[size]
-							? // @ts-expect-error - We check if sizeCallbacks[size] is truthy above
-							  sizeCallbacks[size](slotRenderEndedEvent, advert)
+						sizeCallback !== undefined
+							? sizeCallback(advert, slotRenderEndedEvent)
 							: fastdom.mutate(() => {
 									advert.updateExtraSlotClasses();
 							  }),
 					);
 				}
-				return Promise.resolve(null);
+				return Promise.resolve();
```

Comment:
Should this Promise return a boolean?

```suggestion
				return Promise.resolve(false);
```

Reply:
I think since each of these is a callback it should return `Promise<void>`?

Reply:
That makes sense!

Reply:
I’ll let you make that final choice and adjust L211 accordingly.